platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +179,26 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
+                IOUtils.copy(is, tfbos);
+                try (InputStream inputStream = tfbos.asByteSource()
+                        .openStream()) {
+                    Detector detector = new DefaultDetector();
+                    MediaType mediaType = detector.detect(inputStream, new Metadata());
+
+                    fileExtension = getFileExtensionForMimeType(mediaType.toString()).replace(".",
+                            "");
+                } finally {
+                    is = tfbos.asByteSource()
+                            .openStream();
+                }
+
+            } catch (Exception e) {
+                LOGGER.debug("Failed to guess mimeType for file without extension.");
+            }
+        }
+
```

Comment:
A little further down (again in code you didn't touch), the method `getRootNamespace` is invoked. This class has it's own implementation of the getRootNamespace method, which build the entire DOM. The class ` org.codice.ddf.platform.util.XMLUtils` has a method named `getRootNamespace` that uses the SAX parser and only looks at the first element to get the rootnamespace. Recommend you use that implementation, assuming it does the right thing.